unstructured/file_utils/filetype.py

```diff
@@ -334,6 +334,27 @@ def _is_text_file_a_json(
 ):
     """Detects if a file that has a text/plain MIME type is a JSON file."""
     exactly_one(filename=filename, file=file)
+    if file is not None:
+        file.seek(0)
+        file_content = file.read(4096)
+        if isinstance(file_content, str):
+            file_text = file_content
+        else:
+            file_text = file_content.decode(errors="ignore")
+        file.seek(0)
+    elif filename is not None:
+        with open(filename) as f:
+            file_text = f.read()
```

Comment:
Since this chunk is reused between `_is_text_file_a_csv` and `_is_text_file_a_json`, we could pull that out into a `_read_file_start_for_type_check` function. As noted below, we probably don't need to read the entire file either for CSV or JSON.